### PR TITLE
feat: [CDS-54455]: Support Nexus/Artifactory/Jenkins/Custom Artifact sources in AWS Native Lambda

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
## Describe your changes
Support Nexus/Artifactory/Jenkins/Custom artifact sources 
Test Plan: https://docs.google.com/spreadsheets/d/1oLas59TG6R7mEgFDYPdzmokAkxZNyNpnnPBFInJoGBE/edit?usp=sharing
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44856)

### CODE REVIEW

**Review by rohitk-h — APPROVED** [review]
> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44856)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Comment by rohitk-h** [comment]
Can we add unit tests ?

**Comment by rohitk-h** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java [hunk 1]
```diff
@@ -320,6 +381,7 @@ private void updateFunctionCodeWithArtifact(AwsLambdaArtifactConfig awsLambdaArt
       updateFunctionCodeRequest.dryRun(false).publish(true);
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java:382]: same line is coming twice

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaEntityHelper.java [hunk 2]
```diff
@@ -41,13 +49,20 @@
...
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import javax.annotation.Nonnull;
 
 @Singleton
 @OwnedBy(CDP)
 public class AwsLambdaEntityHelper {
+  private final List<String> NEXUS_PACKAGE_SUPPORTED_TYPES = Arrays.asList("maven", "npm", "nuget", "raw");
```
Comment (sainath-b) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaEntityHelper.java:63]: Can we do separate validation for NEXUS 2 and NEXUS 3?

**Review by piyush-bhuwalka — DISMISSED** [review]
> Reviewed 7 of 9 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44856)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by sainath-b — CHANGES_REQUESTED** [review]
Please address comments.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java (hunk 1 above)
Comment (alluvamsi-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaEntityHelper.java (hunk 2 above)
Comment (alluvamsi-harness) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaEntityHelper.java]: Done

**Review by piyush-bhuwalka — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44856)* status: 5 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 6 of 9 files at r1, 3 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44856)* status: :shipit: complete! ✓

**Pull request merged**